### PR TITLE
Awesome Book Project Functionality and Style add

### PR DESCRIPTION
**Pull request by c00p75**
- Add Functionality added.
- Remove Functionality added.
- LocalStorage used for storing the data user browser.
- HTML5 semantic tag used.
- style add to project.
- Linter Error resolve.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @c00p75 @HidarAhmad37 @Hidar37 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a descriptive readme and also Write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. Kindly Add descriptive README file to your project - please use [this template](https://github.com/microverseinc/readme-template) but remember to customize it to your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+GitHub: @c00p75
+
+Show your support
+Give a ⭐️ if you like this project!
+
+📝 License
+This project is MIT licensed.
+
+NOTE: we recommend using the MIT license - you can set it up quickly by using templates available on GitHub. You can also use any other license if you wish.
```
Comment (Qoosim) [inline README.md]: - [ ] Kindly adhere to the suggestion by the previous reviewer. Kindly use this [new template](https://github.com/microverseinc/readme-template) to update your README file.

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalidated* :red_circle: 

Hi @c00p75, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not use README [new template](https://github.com/microverseinc/readme-template) to make your README file. Kindly use that template to make your README file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md (hunk 1 above)
Comment (mckent05) [reply README.md]: - Please use the new [readme](https://github.com/microverseinc/readme-template) template suggested by the previous reviewers for your project to get reviewed/ approved 🙏 🙏

**Review by mckent05 — COMMENTED** [review]
Hi @c00p75,

# STATUS: INVALID REQUEST ❌ ❌  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that does not use the new microverse readme template. Please fix this 🙏 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+<a name="readme-top"></a>
+
+<div align="center">
+
+  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
+
+</div>
+
+<!-- PROJECT DESCRIPTION -->
```
Comment (Gambit142) [inline README.md]: - [ ] Great work using the recommended template suggested by other reviewers. For this `README.md`, kindly include the **_Table of Contents_** section. **_Table of Contents_** helps users navigate easily thereby creating good User Experience. 👍

File: JS/Script.js [hunk 3]
```diff
@@ -1,73 +1,93 @@
 const bookCollection = document.querySelector('#collection');
 const addBook = document.querySelector('#add-book');
-
```
Comment (SajjadAhmad14) [inline JS/Script.js:3]: This milestone requires to create a class for book objects and add book functionalities only by adding  methods in class.

`
For example: class Books {
  constructor() {
    this.books = [ ];
  }
// A method to add a book to the list
  addBook(title, author) {
   
}
// A method to remove a from from list
  removeBook(bookID) {
    
  }

  // getter method
  getBooks() {
  }
}
`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+In order to run this project you need:
+- Node.js
+- Browser (any)
+- Working PC
+
+### Setup
+
+Clone this repository to your desired folder:
+
```
Comment (Gambit142) [inline README.md]: - [ ] Kindly include instructions for setting up this project on another person's local machine. Instructions should include the repository link for cloning and how to change directory (cd) to the repository

File: JS/Script.js [hunk 5]
```diff
@@ -1,73 +1,93 @@
...
 
 loadBooks();
 
 function updateNewBook() {
-  appendBookHTML(library[library.length-1].title, library[library.length-1].author);
+  appendBookHTML(library[library.length - 1].title, library[library.length - 1].author);
   localStorage.setItem('Awesome Book Collection', JSON.stringify(library));
 }
 
 function removeBook(element) {
```
Comment (SajjadAhmad14) [inline JS/Script.js]: Please add `methods` to `add` and `remove` books in a class and make sure the app functionality works as expected after refactring the code.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
GitHub flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: JS/Script.js [hunk 6]
```diff
@@ -1,73 +1,93 @@
...
+  library = JSON.parse(localStorage.getItem('Awesome Book Collection'));
 }
 
 function storeEntryData(titleElement, authorElement) {
-  let entryObject = {};
+  const entryObject = {};
   entryObject.title = titleElement.value;
   entryObject.author = authorElement.value;
   library.push(entryObject);
 }
```
Comment (Gambit142) [inline JS/Script.js]: - [ OPTIONAL] The `push` method mutates arrays when used. Mutations may lead to unexpected and hard-to-debug issues, where data becomes incorrect somewhere, and you have no idea where it happens. I would advise you to use the `concat` method

**Review by frank1738 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: JS/Script.js [hunk 7]
```diff
@@ -1,73 +1,100 @@
...
-  entryObject.title = titleElement.value;
-  entryObject.author = authorElement.value;
-  library.push(entryObject);
-}
+  storeEntryData = (titleElement, authorElement) => {
+    const entryObject = {};
+    entryObject.title = titleElement.value;
+    entryObject.author = authorElement.value;
+    this.library.push(entryObject);
+  }
```
Comment (Gambit142) [inline JS/Script.js:19]: - [OPTIONAL] I would advise you to use the `concat` method instead `push` method. The push method mutates arrays. Mutations may lead to unexpected and hard-to-debug issues, where data becomes incorrect somewhere, and you have no idea where it happens.

File: JS/Script.js [hunk 8]
```diff
@@ -1,73 +1,100 @@
...
-function updateNewBook() {
-  appendBookHTML(library[library.length-1].title, library[library.length-1].author);
-  localStorage.setItem('Awesome Book Collection', JSON.stringify(library));
+  removeBook = (element) => {
+    const parent = element.parentElement.parentElement;
+    const parrentIndex = Array.from(this.bookCollection.children).indexOf(parent);
+    parent.remove();
+    this.library.splice(parrentIndex, 1);
+    localStorage.setItem('Awesome Book Collection', JSON.stringify(this.library));
+  }
```
Comment (Gambit142) [inline JS/Script.js:58]: - [OPTIONAL] I would advise you to use the `filter` method instead `splice` method. The `splice` method mutates arrays. Mutations may lead to unexpected and hard-to-debug issues, where data becomes incorrect somewhere, and you have no idea where it happens.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Hidar37 @c00p75 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Good UI ✔️ 
- Good pull request summary ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider the optional suggestion I made in the comment under the review. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by DammyShittu** [comment]
Hi @c00p75,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a pull request from the previous milestone kindly open a new pull request for  Awesome books: a complete website with navigation.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**